lib/rules/object-shorthand.js

```diff
@@ -132,12 +134,36 @@ module.exports = {
                 // if we're "never" and concise we should warn now
                 if (APPLY_NEVER && isConciseProperty) {
                     type = node.method ? "method" : "property";
-                    context.report(node, "Expected longform " + type + " syntax.");
+                    context.report({
+                        node: node,
+                        message: "Expected longform " + type + " syntax.",
+                        fix: function(fixer) {
+                            if (node.method) {
+                                if (node.value.generator) {
+                                    return fixer.replaceTextRange([node.range[0], node.key.range[1]], node.key.name + ": function*");
```

Comment:
Noting for other reviewers that the star spacing might be wrong here-- this will be fixed by another pass in auto-fix. This would also be another example of something that could go into a shared repository settings config. (@NickHeiner you can ignore this)